src/coordsim/flow_processors/default_processor.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from coordsim.flow_processors import BaseFlowProcessor
+import numpy as np
+log = logging.getLogger(__name__)
+
+
+class DefaultFlowProcessor(BaseFlowProcessor):
+    """
+    This is the default processor class.
```

Comment:
What does it do?

Reply:
If you could split the `process_flow` into more subfunction or somehow simplify it, it would help understanding and maintaining

Reply:
Here, the simulator calls a single function `process_flow()` of the class. The function should take care of how the processing is handled at a node. The function returns `True` if the flow started processing, and `False` if the flow was dropped. 

I thought at first that the function itself takes care of housekeeping of how metrics are recorded. For example, I have startup delays in SPR, and that requires a different approach than our SWC approach as functions have to be "started" before a flow is processed.

As for modularity, it's up to the function writer how complex the `process_flow()` function becomes. The dev can break it down into more functions if he or she wants. For the default processor, I haven't started with modularization yet. I will break it down into several functions that are called from the `process_flow()` function. E.g., the function will return `True` once the processing started, and then start another simpy process to return used node resources after the flow duration ended.